### PR TITLE
CSV delimiter

### PR DESCRIPTION
**Pull request by alonbrody**
Added the ability to choose the delimiter for CSV files
The default one is still comma

@avinoamr please review

### CODE REVIEW

File: formats/csv.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
 module.exports.type = "csv";
+var DEFAULT_DELIMITER = ','
 
-module.exports.is = function isCSV( sample ) {
-    return sample.toString().split( "\n" )[ 0 ].indexOf( "," ) != -1;
+module.exports.is = function isCSV( sample, delimiter ) {
```
Comment (avinoamr) [inline formats/csv.js]: @alonbrody  The delimiter seems like one out of many possible options, which would make the arguments list too big. Instead, pass in an options object, that may have a `delimiter` key.

**Comment by avinoamr** [comment]
@alonbrody I'm not sure about that... Structurize should be able to identify the type automatically. Passing in a delimiter doesn't make sense to me, because it means that the user must already know that he has a CSV, in which case using structurize doesn't make sense. For parsing it's fine, but not for sampling. How do you intend to use it?

**Comment by alonbrody** [comment]
@avinoamr I agree that guessing if it's a CSV or not is redundant and I will remove it. But we still need to parse the CSV itself so we can use the structurize.parser to do so like I did [here](https://github.com/avinoamr/color/pull/671) (ignore the tests that failed because it depends on this pr)

This way we will be able to ingest all value separated files that the users will try (as long as they will state the delimiter). I couldn't find a real way to properly identify the delimiter. So I gave the user the option to choose the delimiter.

File: formats/csv.js (hunk 1 above)
Comment (alonbrody) [inline formats/csv.js]: Agree, will change the whole thing to accept options instead of specific parameter

**Comment by avinoamr** [comment]
@alonbrody I don't see how it's going to work. The file won't be identified as a CSV, because no comma will be found by the type guesser. No?

**Comment by alonbrody** [comment]
Unless you will tell the structurize the actual delimiter of the file. In this case there's no need to guess and we can start parsing based on this delimiter directly.

**Comment by alonbrody** [comment]
@avinoamr Added another commit. Just in case you missed that

**Comment by avinoamr** [comment]
@alonbrody Merged.

**Comment by avinoamr** [comment]
Published as version 1.2.6

**Pull request merged**